### PR TITLE
Test for commands: object_tree, new_attr, sprintf

### PR DESCRIPTION
**Pull request by t-wissmann**
This adds a couple of more tests for the meta commands object_tree,
new_attr, and sprintf. This mainly concerns error handling and
completion. This should improve the test coverage.

### CODE REVIEW

**Review by blarz — DISMISSED** [review]
It improves coverage indeed!

**Comment by t-wissmann** [comment]
It turned out a couple of error messages had to be moved to the error channel.

**Review by blarz — APPROVED** [review]

File: src/metacommands.cpp [hunk 1]
```diff
@@ -393,7 +393,7 @@ static std::map<string, function<Attribute*(string)>> name2constructor {
 Attribute* MetaCommands::newAttributeWithType(string typestr, string attr_name, Output output) {
     auto it = name2constructor.find(typestr);
     if (it == name2constructor.end()) {
-        output << "error: unknown type \"" << typestr << "\"";
+        output.perror() << "error: unknown type \"" << typestr << "\"";
```
Comment (blarz) [inline src/metacommands.cpp:396]: "error:" can probably be removed?

**Review by blarz — CHANGES_REQUESTED** [review]

File: src/metacommands.cpp (hunk 1 above)
Comment (blarz) [reply src/metacommands.cpp:396]: (Which probably is the reason for the merge conflict)

Comment (t-wissmann) [reply src/metacommands.cpp]: Indeed, thanks! (it's resolved now)

Comment (blarz) [reply src/metacommands.cpp]: Thanks!

**Review by blarz — APPROVED** [review]

**Pull request merged**